### ISSUE TITLE
multiarch build works for amd64 and arm64 but not for arm/v7 ("qemu-arm: Could not open '/lib/ld-linux.so.3'")

### ISSUE THREAD

**Issue by hamishcunningham** [docker/buildx#1222]
Hello!

[This Dockerfile](https://gitlab.com/hamishcunningham/the-internet-of-things/-/blob/master/support/Dockerfile) when run using 
```bash
docker buildx --platform linux/arm/v7,linux/arm64/v8,linux/amd64  .
```
fails on the arm/v7 build with this error:
```
qemu-arm: Could not open '/lib/ld-linux.so.3': No such file or directory
```
Various similar reports imply that linking the .so from a package called `libc6-armhf-cross` might help, but I couldn't get that to work.

I'm building on Ubuntu 22.04 running on x86_64 hardware.

Any ideas? Thanks!

**Comment by crazy-max** [comment]
Can't repro the QEMU on my side but Dockerfile fails to build:

```console
$ docker buildx build --platform linux/arm/v7,linux/arm64/v8,linux/amd64 .
#64 [linux/arm/v7 19/27] RUN . .bashrc && cd the-internet-of-things/exercises/HelloWorld &&   /home/ubuntu/the-internet-of-things/support/tooling/platformio/penv/bin/pio run
#64 95.87 Downloading
#64 96.15 Unpacking
#64 96.53 Tool Manager: tool-scons@4.40300.220706 has been installed!
#64 98.85 Verbose mode can be enabled via `-v, --verbose` option
#64 98.93 CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/featheresp32.html
#64 98.93 PLATFORM: Espressif 32 (4.4.0+sha.7ce076f) > Adafruit ESP32 Feather
#64 98.93 HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
#64 98.93 DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
#64 98.93 PACKAGES:
#64 98.93  - framework-arduinoespressif32 @ 2.0.4+sha.5131b87
#64 98.93  - tool-esptoolpy @ 1.30300.0 (3.3.0)
#64 98.93  - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch3
#64 98.94 Converting sketch.ino
#64 98.97 qemu-arm: Could not open '/lib/ld-linux.so.3': No such file or directory
#64 98.97 *** Error 255
#64 98.98 AssertionError: :
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/penv/lib/python3.10/site-packages/platformio/builder/main.py", line 188:
#64 98.98     env.SConscript("$BUILD_SCRIPT")
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/packages/tool-scons/scons-local-4.3.0/SCons/Script/SConscript.py", line 597:
#64 98.98     return _SConscript(self.fs, *files, **subst_kw)
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/packages/tool-scons/scons-local-4.3.0/SCons/Script/SConscript.py", line 285:
#64 98.98     exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/platforms/espressif32/builder/main.py", line 283:
#64 98.98     target_elf = env.BuildProgram()
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/packages/tool-scons/scons-local-4.3.0/SCons/Util.py", line 742:
#64 98.98     return self.method(*nargs, **kwargs)
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/penv/lib/python3.10/site-packages/platformio/builder/tools/platformio.py", line 62:
#64 98.98     env.ProcessProgramDeps()
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/packages/tool-scons/scons-local-4.3.0/SCons/Util.py", line 742:
#64 98.98     return self.method(*nargs, **kwargs)
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/penv/lib/python3.10/site-packages/platformio/builder/tools/platformio.py", line 122:
#64 98.98     env.BuildFrameworks(env.get("PIOFRAMEWORK"))
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/packages/tool-scons/scons-local-4.3.0/SCons/Util.py", line 742:
#64 98.98     return self.method(*nargs, **kwargs)
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/penv/lib/python3.10/site-packages/platformio/builder/tools/platformio.py", line 325:
#64 98.98     env.ConvertInoToCpp()
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/packages/tool-scons/scons-local-4.3.0/SCons/Util.py", line 742:
#64 98.98     return self.method(*nargs, **kwargs)
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/penv/lib/python3.10/site-packages/platformio/builder/tools/pioino.py", line 240:
#64 98.98     out_file = c.convert(ino_nodes)
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/penv/lib/python3.10/site-packages/platformio/builder/tools/pioino.py", line 88:
#64 98.98     return self.process(contents)
#64 98.98   File "/home/ubuntu/the-internet-of-things/support/tooling/platformio/penv/lib/python3.10/site-packages/platformio/builder/tools/pioino.py", line 109:
#64 98.98     assert self._gcc_preprocess(contents, out_file)
#64 99.10 ========================= [FAILED] Took 97.57 seconds =========================
#64 ERROR: process "/bin/bash -c . .bashrc && cd the-internet-of-things/exercises/HelloWorld &&   ${PIO_CLI} run" did not complete successfully: exit code: 1
```

I guess you have followed https://github.com/docker/buildx#building-multi-platform-images?, specially:

```console
$ docker run --privileged --rm tonistiigi/binfmt --install all
```

If you still have the issue can you try https://github.com/docker/buildx/issues/1210#issuecomment-1185897093

**Comment by crazy-max** [comment]
Oh nevermind:

```
#64 98.97 qemu-arm: Could not open '/lib/ld-linux.so.3': No such file or directory
```

**Comment by crazy-max** [comment]
Ok so this is a linker issue in your Dockerfile and not a QEMU or BuildKit issue to me.

**Comment by hamishcunningham** [comment]
Yes, I did this to upgrade buildx:
```
  docker buildx create --name mybuilder
  docker buildx use mybuilder
  docker run --privileged --rm tonistiigi/binfmt --install \
    linux/arm/v7,linux/arm64/v8,linux/amd64
  docker buildx inspect --bootstrap
```

I did think that perhaps this was caused by ubuntu dropping 32 bit support but they do seem to support it on Raspberry Pi so I'm still a little puzzled. Given that it is buildx that calls qemu-arm it seems that I can't really address the issue, which is why I raised it here, but it isn't critical for me at present so I'm fine to close it if you prefer.

Tnx!
